File: language-samples/hello.r
# R sample for GitHub language detection
# BarbrickDesign - Complete Language Portfolio

Greeter <- setRefClass("Greeter",
  fields = list(message = "character"),
  methods = list(
    greet = function() {
      cat(message, "\n")
    }
  )
)

main <- function() {
  greeter <- Greeter$new(message = "Hello from R!")
  greeter$greet()
  
  # Vector
  numbers <- c(1, 2, 3, 4, 5)
  doubled <- numbers * 2
  
  # List
  info <- list(
    language = "R",
    paradigm = "Multi-paradigm",
    typing = "Dynamic"
  )
  
  # Apply functions
  languages <- c("R", "Python", "Julia")
  upper <- sapply(languages, toupper)
  
  # Data frame
  df <- data.frame(
    name = c("R", "Python", "Julia"),
    year = c(1993, 1991, 2012)
  )
  
  print(df)
  print(upper)
}

main()

File: language-samples/hello.jl
# Julia sample for GitHub language detection
# BarbrickDesign - Complete Language Portfolio

struct Greeter
    message::String
end

function greet(g::Greeter)
    println(g.message)
end

function main()
    greeter = Greeter("Hello from Julia!")
    greet(greeter)
    
    # Array
    numbers = [1, 2, 3, 4, 5]
    doubled = [x * 2 for x in numbers]
    
    # Dictionary
    info = Dict(
        "language" => "Julia",
        "paradigm" => "Multi-paradigm",
        "typing" => "Dynamic"
    )
    
    # Map function
    languages = ["julia", "python", "r"]
    upper = map(uppercase, languages)
    
    # Broadcasting
    result = numbers .* 2
    
    println(doubled)
    println(upper)
    println(info)
end

main()
